stetho/src/main/java/com/facebook/stetho/common/android/HandlerUtil.java

```diff
@@ -11,38 +11,104 @@
 
 import android.os.Handler;
 import android.os.Looper;
+import com.facebook.stetho.common.Util;
+
+import java.util.concurrent.Callable;
 
 public final class HandlerUtil {
   private HandlerUtil() {
   }
 
-  public static boolean postAndWait(Handler handler, final Runnable r) {
-    if (Looper.myLooper() == handler.getLooper()) {
-      r.run();
-      return true;
+  /**
+   * Checks whether the current thread is the same thread that the {@link Handler} is associated
+   * with.
+   * @return true if the current thread is the same thread that the {@link Handler} is associated
+   * with; otherwise false.
+   */
+  public static boolean checkAccess(Handler handler) {
```

Comment:
Strangely named method IMO.  Kind of a strange thing to expose publicly as well.  It's not significantly less code than writing the below expression and it's not more clear than the expression...